cmd/everoute-agent/main.go

```diff
@@ -134,15 +136,22 @@ func initCNI(datapathManager *datapath.DpManager, mgr manager.Manager, proxySync
 			overlaySyncChan <- ersource.NewReplayEvent()
 		}
 		datapathManager.SetOverlaySyncFunc(overlayReplayFunc)
+	} else if opts.UseEverouteIPAM() {
+		klog.Fatalf("can't use everoute ipam if disable overlay")
 	}
+
 	if opts.IsEnableProxy() {
 		proxySyncFunc := func() {
 			proxySyncChan <- ersource.NewReplayEvent()
 		}
 		datapathManager.SetProxySyncFunc(proxySyncFunc)
 	}
 
-	setAgentConf(datapathManager, mgr.GetAPIReader())
+	c, err := client.New(mgr.GetConfig(), client.Options{Scheme: clientsetscheme.Scheme})
+	if err != nil {
+		klog.Fatalf("Failed to new a client, err: %v", err)
+	}
+	setAgentConf(datapathManager, c)
```

Comment:
what is the difference between `New Client` and `mgr.GetAPIReader()`

Reply:
when use everoute ipam, It will request ippool to allocate gateway ip, and client used to update ippool status